float32[12] kp
float32[12] kd
bool[12] mask